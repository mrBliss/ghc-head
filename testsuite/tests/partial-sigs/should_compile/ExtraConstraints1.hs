{-# LANGUAGE PartialTypeSignatures #-}
module ExtraConstraints1 where

arbitCs1 :: _ => a -> String
arbitCs1 x = show (succ x) ++ show (x == x)

arbitCs2 :: (Show a, _) => a -> String
arbitCs2 = arbitCs1

arbitCs3 :: (Show a, Enum a, _) => a -> String
arbitCs3 = arbitCs1

arbitCs4 :: (Eq a, _) => a -> String
arbitCs4 = arbitCs1

arbitCs5 :: (Eq a, Enum a, Show a, _) => a -> String
arbitCs5 = arbitCs1
