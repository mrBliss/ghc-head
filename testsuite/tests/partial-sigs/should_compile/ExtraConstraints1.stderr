TYPE SIGNATURES
  arbitCs1 :: forall a. (Show a, Eq a, Enum a) => a -> String
  arbitCs2 :: forall a. (Show a, Eq a, Enum a) => a -> String
  arbitCs3 :: forall a. (Show a, Enum a, Eq a) => a -> String
  arbitCs4 :: forall a. (Eq a, Show a, Enum a) => a -> String
  arbitCs5 :: forall a. (Eq a, Enum a, Show a) => a -> String
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
